### PR TITLE
Fix compiler warnings

### PR DESCRIPTION
**Pull request by davydden**
fixes https://github.com/elemental/Elemental/issues/243 (most of them)
~~I will double check tomorrow on Ubuntu that this indeed works (thus WIP).~~ builds on Ubuntu+gcc 5.4.0.

@poulson ping.

### CODE REVIEW

File: include/El/core/imports/choice.hpp [hunk 1]
```diff
@@ -86,34 +86,35 @@ class Args
     vector<bool> usedArgs_;
     ostream& error_;
 
-    virtual void HandleVersion( ostream& os=cout ) const { }
-    virtual void HandleBuild( ostream& os=cout ) const { }
+    virtual void HandleVersion( ostream& os=cout ) const { (void)os; }
```
Comment (davydden) [inline include/El/core/imports/choice.hpp]: p.s. one could as well had those `inline void Args::...` within the header below, the current diff is just smaller.

Comment (poulson) [reply include/El/core/imports/choice.hpp]: I think defining an `EL_UNUSED` macro ala https://stackoverflow.com/a/1486931 would be more readable than the explicit voiding. Would you mind adding such a macro? This PR should be ready after that.

Comment (poulson) [reply include/El/core/imports/choice.hpp]: Also, thanks!

Comment (davydden) [reply include/El/core/imports/choice.hpp]: where do keep macros? where shall I put it?

Comment (poulson) [reply include/El/core/imports/choice.hpp]: I think in include/El/core.hpp would be most consistent.

**Comment by davydden** [comment]
@poulson ping.

**Comment by poulson** [comment]
There was a problem with Travis CI's backend so I just manually restarted the CI build.

**Pull request merged**